platform-sdk/docs/proposals/TSS-Ledger-Id/tss-ledger-id.md

```diff
@@ -0,0 +1,73 @@
+## Interactions
+
+| Interaction       | Entities                     | 
+|-------------------|------------------------------|
+| Designers         | Edward, Kore, Anthony, Artem | 
+| Implementers      | team-spacetime               |
+| Team Impacts      | Services, DevOps             |
+| Related Proposals | TSS-Library, TSS-Block-Proof |
+| HIPS              | HIP-???                      |
+
+# Context
+
+To support the ability to so block proofs, we need the ability to assign a public/private key-pair to the network or ledger itself.  The following requirements on the ledger key must hold: 
+* The public key of the ledger must not change when the composition of network nodes change. 
+* The private key of the ledger must not be known by anyone. 
+* The signature of the ledger private key signing a message must be producible from a threshold of network nodes signing the message. 
+* Verification of the ledger signature must be cheap for an EVM smart contract. 
+
+## Design Decisions
+
+Describe the high level options available, the trade-offs between them and the decisions made to commit to one option
+and its opportunity costs.
+
+# Changes
+
+## Public API
+
+Describe any public API changes or additions. Include stakeholders of the API. Mark this section as N/A if not
+applicable.
+
+Code can be included in the proposal directory, but not submitted to the code base.
+
+## Core Behaviors
+
+Describe any new or modified behavior. Mark this section as N/A if not applicable.
+
+### Metrics
+
+How is the behavior observable in metrics?
+
+## Performance
+
+Describe any expected performance changes. This section is mandatory for platform wiring changes. Mark this section as
+N/A if not applicable.
+
+## Interactions
+
+Describe any interactions with other major systems, teams, or proposals. Mark this section as N/A if not applicable.
+
+## Modules and Repos
+
+Describe any new or modified modules or repositories. Mark this section as N/A if not applicable.
+
+# Test Plan
+
+## Unit Tests
+
+Describe any unit tests needed, or delete this section if not applicable.
+
+## Integration Tests
+
+Describe any integration tests needed, or delete this section if not applicable. Integration tests include migration,
+reconnect, restart, etc.
+
+## Performance Tests
+
+Describe any performance tests needed, or delete this section if not applicable.
+
+Performance tests include high TPS, specific work loads that stress test the system, or longevity tests.
+
+# Implementation and Delivery Plan
+
+Describe the implementation and delivery plan for the capability described in the proposal.   What does the order of implementation of pieces need to be?   What are the stages or phases needed for the delivery of capabilities?
```

Comment:
I again disagree on that point. Once we agree on the proposal, an epic issue will be created that will split up into sub-issues that might even change over time. In the proposal you often do not have that knowledge

Reply:
Then the proposal is not thought out enough.

Reply:
There are capabilities that require a multi-stage release. 
1. Prep the software and state for a migration 
2. Perform the migration
3. Cleanup from the migration. 

There are capabilities the would benefit from staged rollout just for integrity checks. 
1. TSS 1 - give all nodes EC Keys
2. TSS 2 - If there are no problems...  Rollout the TSS code to perform keying of shares.